src/serde/de_br.rs

```diff
@@ -156,11 +155,7 @@ pub fn traverse_path_with_vec(
         if parsing_sexp {
             match allocator.sexp(sexp_to_parse) {
                 SExp::Atom => {
-                    return Err(EvalErr(
-                        sexp_to_parse,
-                        "invalid backreference during deserialisation".into(),
-                    )
-                    .into());
+                    return Err(EvalErr::SerializationError);
```

Comment:
I think it would be helpful to have its own error code for this too actually. having a back-reference that points into something invalid in the tree is different from bad encoding.

Reply:
switched